File: python/word-count/word_count.py
import re

def count_words(sentence):

    output = {}

    sentence = re.split(',| |\t|\n|_', sentence)

    for word in sentence:
        # remove excess stuff
        word = word.strip('\"')
        word = word.strip("\'")
        word = re.sub(r'[!&@$%^&,.:]','',word)




        if(word == ''):
            continue

        # to lower
        lowerWord = word.lower()

        if lowerWord in output:
            output[lowerWord] += 1
        else:
            output [lowerWord] = 1


    return output

File: javascript/word-count/word-count.js
// not really happy with this, but I need better regex understanding to improve this
export const countWords = (phrase) => {

  let output = {};


  let words = phrase.split(/[\s,]+/);

  words.forEach((word) => {

    if(word == ''){
      return;
    }

    word = word.toLowerCase().replace(/[.,\/#!@$%\^&\*;:{}=\-_`~()]/g,"").trim("'");
    word = word.replace(/^'/, '');
    word = word.replace(/'$/, '');




    if(word in output){
      output[word]++;
    }
    else{
      output[word] = 1;
    }
  });


  return output;

};
